src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,257 @@
+
+    @Override
+    public Car update(Car car) {
+        String selectQuery = "UPDATE cars SET model = ?, manufacturer_id = ? WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            Car old = get(car.getId()).get();
+            preparedStatement.executeUpdate();
+            updateDriversByCar(car);
+            return old;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE cars SET deleted = true WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                             selectQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT cars.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars"
+                + " LEFT JOIN manufacturers m on cars.manufacturer_id = m.id"
+                + " join cars_drivers cd on cars.id = cd.car_id"
+                + " join drivers d on cd.driver_id = d.id"
+                + " where cars.deleted = false and driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    private void updateDriversByCar(Car car) {
+        insertAllDrivers(car.getId(), car.getDrivers());
+        deleteAllDriversExceptList(car.getId(), car.getDrivers());
+    }
+
+    private void insertAllDrivers(Long carId, List<Driver> drivers) {
```

Comment:
Im not sure you need to complicate it this much, you don't need to delete or insert ALL drivers, you can do it with a simple query in respective methods, such as `DELETE FROM cars_drivers WHERE car_id = ?`